File: src/main/java/org/minturtle/careersupport/interview/entity/InterviewTemplate.java
package org.minturtle.careersupport.interview.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.Instant;
import java.time.ZonedDateTime;

@Document(collection = "interview_templates")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@Getter
public class InterviewTemplate {

    @Id
    private String id;
    private String userId;
    private String theme;

    @Builder.Default
    private Instant createdAt = Instant.now();
}

File: src/main/kotlin/org/minturtle/careersupport/interview/entity/InterviewTemplate.kt
package org.minturtle.careersupport.interview.entity

import org.minturtle.careersupport.common.utils.NanoIdGenerator
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document(collection = "interview_templates")
class InterviewTemplate(
    @Id
    var id: String = NanoIdGenerator.createNanoId(idSize = 10),
    val userId: String,
    val theme: String,
    val createdAt: Instant = Instant.now()
)